Mathlib/CategoryTheory/Yoneda.lean

```diff
@@ -543,4 +544,178 @@ lemma isIso_of_yoneda_map_bijective {X Y : C} (f : X ⟶ Y)
+  evaluationUncurried C (Type v₁) ⋙ uliftFunctor.{u₁}
+
+@[simp]
+theorem coyonedaEvaluation_map_down (P Q : C × (C ⥤ Type v₁)) (α : P ⟶ Q)
+    (x : (coyonedaEvaluation C).obj P) :
+    ((coyonedaEvaluation C).map α x).down = α.2.app Q.1 (P.2.map α.1 x.down) :=
+  rfl
+
+/-- The "Coyoneda pairing" functor, which sends `X : C` and `F : C ⥤ Type`
+to `coyoneda.rightOp.obj X ⟶ F`, functorially in both `X` and `F`.
+-/
+def coyonedaPairing : C × (C ⥤ Type v₁) ⥤ Type max u₁ v₁ :=
+  Functor.prod coyoneda.rightOp (𝟭 (C ⥤ Type v₁)) ⋙ Functor.hom (C ⥤ Type v₁)
+
+-- Porting note: we need to provide this `@[ext]` lemma separately,
+-- as `ext` will not look through the definition.
+-- See https://github.com/leanprover-community/mathlib4/issues/5229
+@[ext]
+lemma coyonedaPairingExt {X : C × (C ⥤ Type v₁)} {x y : (coyonedaPairing C).obj X}
+    (w : ∀ Y, x.app Y = y.app Y) : x = y :=
+  NatTrans.ext _ _ (funext w)
+
+@[simp]
+theorem coyonedaPairing_map (P Q : C × (C ⥤ Type v₁)) (α : P ⟶ Q) (β : (coyonedaPairing C).obj P) :
+    (coyonedaPairing C).map α β = coyoneda.map α.1.op ≫ β ≫ α.2 :=
+  rfl
+
+universe w in
+variable {C} in
+/-- A bijection `(coyoneda.obj X ⋙ uliftFunctor ⟶ F) ≃ F.obj (unop X)` which is a variant
+of `coyonedaEquiv` with heterogeneous universes. -/
+def coyonedaCompUliftFunctorEquiv (F : C ⥤ Type max v₁ w) (X : Cᵒᵖ) :
+    (coyoneda.obj X ⋙ uliftFunctor.{w} ⟶ F) ≃ F.obj X.unop where
+  toFun φ := φ.app X.unop (ULift.up (𝟙 _))
+  invFun f :=
+    { app := fun Y x => F.map (ULift.down x) f }
+  left_inv φ := by
+    ext Y f
+    dsimp
+    rw [← FunctorToTypes.naturality]
+    dsimp
+    rw [Category.id_comp]
+    rfl
+  right_inv f := by aesop_cat
+
+/-- The Coyoneda lemma asserts that the Coyoneda pairing
+`(X : C, F : C ⥤ Type) ↦ (coyoneda.obj X ⟶ F)`
+is naturally isomorphic to the evaluation `(X, F) ↦ F.obj X`.
+
+See <https://stacks.math.columbia.edu/tag/001P>.
+-/
+def coyonedaLemma : coyonedaPairing C ≅ coyonedaEvaluation C :=
+  NatIso.ofComponents
+    (fun X ↦ Equiv.toIso (coyonedaEquiv.trans Equiv.ulift.{u₁, v₁}.symm))
+    (by intro (X, F) (Y, G) f
+        ext (a : coyoneda.obj (op X) ⟶ F)
+        apply ULift.ext.{u₁, v₁}
+        simp only [Functor.prod_obj, Functor.id_obj, types_comp_apply, coyonedaEvaluation_map_down]
+        erw [Equiv.ulift_symm_down, Equiv.ulift_symm_down]
+        simp [coyonedaEquiv, ← FunctorToTypes.naturality])
+
+variable {C}
+
+/-- The isomorphism between `coyoneda.obj X ⟶ F` and `F.obj (unop X)`
+(we need to insert a `ULift` to get the universes right!)
+given by the Coyoneda lemma.
+-/
+@[simps!]
+def coyonedaSections (X : Cᵒᵖ) (F : C ⥤ Type v₁) :
+    (coyoneda.obj X ⟶ F) ≅ ULift.{u₁} (F.obj X.unop) :=
+  (coyonedaLemma C).app (X.unop, F)
+
+/-- When `C` is a small category, we can restate the isomorphism from `coyonedaSections`
+without having to change universes.
+-/
+def coyonedaSectionsSmall {C : Type u₁} [SmallCategory C] (X : Cᵒᵖ) (F : C ⥤ Type u₁) :
```

Comment:
If seems that this definition can be removed. Could you try to get rid of  `coyonedaSections` as well? (Currently, it is certainly unused! But, for the other, as this is a direct application of `yonedaLemma`, it should be possible to change the downstream uses of it so that it refers directly to the (refactored) `yonedaLemma`.)

Reply:
I removed them.